dw-dialog.js

```diff
@@ -381,6 +383,17 @@ export const DwModalDialogMixin = (baseElement) => class DwModalDialog extends b
     this.dispatchEvent(event);
     this._unlistenEvents();
 
+    const index = window.openedDwDialogsInstances.findIndex((element) => element === this);
+    if (index >= 0) {
+      window.openedDwDialogsInstances.splice(index, 1);
+    }
+
+    window.openedDwDialogsInstances.map((element, index) => {
```

Comment:
Same as here, we don't need to iterate over the list and check for the last index. Instead, we can just set the `escapeKeyAction` property of the last element directly.

Note: Even if we need iteration over a list, we shouldn't use `array.map` as it'll return a new array. Instead, we should use `array.forEach`. It can be used only if we needs to build new array from existing array.